include/gs_types.h

```diff
@@ -162,11 +194,15 @@ namespace gs
         Loc1 scale;
         Boolean active;
         std::optional<ObjectID> parent;
+
+        bool operator==(const Object1 &other) const = default;
```

Comment:
Is there a reason this one (and a couple of others) use `operator==` and the most use `operator<=>`?

Reply:
I had some issues with `<=>` not being defined on `std::optional` or `std::vector` (which might vary by implementation), so for the types using those I just stuck with the equality that I needed.